Source/MQTTnet.AspnetCore/ReaderExtensions.cs

```diff
@@ -81,23 +83,24 @@ static bool TryReadBodyLength(ref ReadOnlySequence<byte> input, out int headerLe
             var index = 1;
             headerLength = 0;
             bodyLength = 0;
-            
-            var temp = input.Slice(0, Math.Min(5, input.Length)).GetMemory().Span;
+
+            var valueSequence = input.Slice(0, Math.Min(5, input.Length));
+            var valueSpan = valueSequence.IsSingleSegment ? valueSequence.First.Span : valueSequence.ToArray();
```

Comment:
`valueSequence.Length == 5` so rather short, and lets assume that `valueSequence.IsSingleSegment == true / false` is both likely (though == true will be more likely) so `ToArray` will be expensive.

```c#
Span<byte> valueSpan = stackalloc byte[8];    // 8 as next power of 2
valueSequence.CopyTo(valueSpan);
valueSpan = valueSpan.Slice(0, valueSequence.Length);
```

`FirstSpan` is also not really cheap, so I'd just straighten the code and copy always. It's only a few bytes, but the code is much shorter so it should outweight the cost of copying.

Reply:
Benefit a lot！